test/utils.ts

```diff
@@ -1,6 +1,8 @@
 import { expect } from 'bun:test'
 import type { Tensor } from '@shumai/shumai'
 
+export type ArrayLike = Float32Array | Float64Array | Uint8Array | number[]
```

Comment:
not necessary in this PR, but we should maybe expose types like this in `shumai/util/types.ts`.  `sm.util.viter` also takes an "ArrayLike"

Reply:
I was actually thinking the same thing while working on this PR and agree 100%!

Reply:
What other typed arrays will the project ultimately want to include here beyond those already listed? (Assuming you're okay with this being added under this current PR)

Reply:
I think BigInt64 and Int32